docs/DeveloperGuide.md

```diff
@@ -200,6 +200,27 @@ allow subclasses of `Tag` to be tagged to a person. Currently, there are 4 of su
 
 ![Class diagram for Tag](images/TagClassDiagram.png)
 
+Given below is an example usage scenario and how the tagging mechanism behaves at each step.
+
+Step 1. The user enters the following valid `TagCommand`: `tag 1 edu/computer science m/cs2030s m/cs2040s` and `LogicManager`
+would execute it.
+
+Step 2. `LogicManger` would pass the argument to `AddressBookParser` to parse the command and identify it as a `TagCommand`.
+It will then pass the arguments to `TagCommandParser` to handle the parsing for the identified `TagCommand`.
+
+Step 3. `TagCommandParser` would first parse the index using `ParserUtil#parseIndex()` to identify the person to tag to.
+Afterwards, `TagCommandParser` would parse the tag arguments provided using `ParserUtil#parseTags()` to identify the individual
+tag types for the arguments provided. 
+
+Step 4. After parsing the arguments, the control is handed over to `TagCommand` where it will return a `TagCommand` object. It
+will eventually return to `LogicManager` which will call `TagCommand#execute()` to execute the command.
+
+Step 5. Upon execution, the person will be fetched and tagged using `Model#setPerson`. The edited person would then be updated
+and stored in the addressbook.`CommandResult` would then generate a success message to inform the user the person has been tagged
+successfully.
+
+![The following sequence diagram shows how the tag operation works:](images/TagSequenceDiagram.png)
+
```

Comment:
Good job! Looks clear and very readable. I should adapt mine to this style as well.